plugins/integration/lib/events/kIntegrationFlowManager.php

```diff
@@ -39,7 +41,16 @@ public function shouldConsumeJobStatusEvent(BatchJob $dbBatchJob)
 	public static function addintegrationJob($objectType, $objectId, kIntegrationJobData $data) 
 	{
 		$partnerId = kCurrentContext::getCurrentPartnerId();
-		
+	
+		$providerType = $data->getProviderType();
+		$integrationProviderObj = KalturaPluginManager::loadObject('IIntegrationProvider', $providerType);
+
+		if($integrationProviderObj && !$integrationProviderObj->validatePermissions($partnerId))
```

Comment:
Please change to if(!$integrationProviderObj || !$integrationProviderObj->validatePermissions($partnerId)).
Also, please rename $integrationProviderObj to $integrationProvider.